app/recurringjob/type.go

```diff
@@ -18,7 +18,7 @@ type Job struct {
 	lhClient *lhclientset.Clientset        // Kubernetes clientset for Longhorn resources.
 
 	eventRecorder record.EventRecorder // Used to record events related to the job.
-	logger        logrus.FieldLogger   // Log messages related to the job.
+	logger        *logrus.Logger       // Log messages related to the job.
```

Comment:
What's the differences compared to the original logrus.FieldLogger?

Reply:
`*logrus.Logger` is the base logger instance, while `logrus.FieldLogger` is an interface implemented by the base logger instance type.

In our case, goroutines invoke `.Info()` on `*logrus.Entry` (via the `FieldLogger` interface), which leads to a panic due to concurrent access to the underlying logger's mutex.

https://github.com/longhorn/longhorn-manager/blob/v1.9.0/vendor/github.com/sirupsen/logrus/entry.go#L233-L236
https://github.com/longhorn/longhorn-manager/blob/v1.9.0/vendor/github.com/sirupsen/logrus/logger.go#L225-L227